File: braggbnb107_ui/src/services/IdentityVerificationService.js
import http from "../http-common"; 

class IdentityVerificationService {
  getAllIdentityVerifications(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/identityVerification/identityVerifications`, searchDTO);
  }

  get(identityVerificationId) {
    return this.getRequest(`/identityVerification/${identityVerificationId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/identityVerification?field=${matchData}`, null);
  }

  addIdentityVerification(data) {
    return http.post("/identityVerification/addIdentityVerification", data);
  }

  update(data) {
  	return http.post("/identityVerification/updateIdentityVerification", data);
  }
  
  uploadImage(data,identityVerificationId) {
  	return http.postForm("/identityVerification/uploadImage/"+identityVerificationId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new IdentityVerificationService();

File: src/main/java/com/braggbnb107/service/IdentityVerificationService.java
package com.braggbnb107.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb107.domain.IdentityVerification;
import com.braggbnb107.dto.IdentityVerificationDTO;
import com.braggbnb107.dto.IdentityVerificationSearchDTO;
import com.braggbnb107.dto.IdentityVerificationPageDTO;
import com.braggbnb107.dto.IdentityVerificationConvertCriteriaDTO;
import com.braggbnb107.service.GenericService;
import com.braggbnb107.dto.common.RequestDTO;
import com.braggbnb107.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface IdentityVerificationService extends GenericService<IdentityVerification, Integer> {

	List<IdentityVerification> findAll();

	ResultDTO addIdentityVerification(IdentityVerificationDTO identityVerificationDTO, RequestDTO requestDTO);

	ResultDTO updateIdentityVerification(IdentityVerificationDTO identityVerificationDTO, RequestDTO requestDTO);

    Page<IdentityVerification> getAllIdentityVerifications(Pageable pageable);

    Page<IdentityVerification> getAllIdentityVerifications(Specification<IdentityVerification> spec, Pageable pageable);

	ResponseEntity<IdentityVerificationPageDTO> getIdentityVerifications(IdentityVerificationSearchDTO identityVerificationSearchDTO);
	
	List<IdentityVerificationDTO> convertIdentityVerificationsToIdentityVerificationDTOs(List<IdentityVerification> identityVerifications, IdentityVerificationConvertCriteriaDTO convertCriteria);

	IdentityVerificationDTO getIdentityVerificationDTOById(Integer identityVerificationId);







}





